server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexCannedResponse.scala

```diff
@@ -0,0 +1,81 @@
+package uk.gov.ons.addressIndex.server.modules
+
+import uk.gov.ons.addressIndex.crfscala.CrfScala.CrfTokenResult
+import uk.gov.ons.addressIndex.model.server.response._
+
+trait AddressIndexCannedResponse {
+
+  def Limit: Int = 10
+
+  def searchContainerTemplate(
+    tokens: Seq[CrfTokenResult],
+    addresses: Seq[AddressResponseAddress],
+    total: Int
+  ): AddressBySearchResponseContainer = {
+    AddressBySearchResponseContainer(
+      response = AddressBySearchResponse(
+        tokens = tokens.groupBy(_.label).map(x => x._1 -> x._2.map(_.value).mkString(" ")).map(p => CrfTokenResult(p._2, p._1)).toSeq,
+        addresses = addresses,
+        limit = Limit,
+        offset = 0,
+        total = addresses.size
+      ),
+      status = OkAddressResponseStatus
+    )
+  }
+
+  def searchUprnContainerTemplate(optAddresses: Option[AddressResponseAddress]): AddressByUprnResponseContainer = {
+    AddressByUprnResponseContainer(
+      response = AddressByUprnResponse(
+        address = optAddresses
+      ),
+      status = OkAddressResponseStatus
+    )
+  }
+
+  def NoAddressFoundUprn: AddressByUprnResponseContainer = {
+    AddressByUprnResponseContainer(
+      response = AddressByUprnResponse(
+        address = None
+      ),
+      status = NotFoundAddressResponseStatus,
+      errors = Seq(NotFoundAddressResponseError)
+    )
+  }
+
+  def UnsupportedFormatUprn: AddressByUprnResponseContainer = {
+    AddressByUprnResponseContainer(
+      response = AddressByUprnResponse(
+        address = None
+      ),
+      status = BadRequestAddressResponseStatus,
+      errors = Seq(FormatNotSupportedAddressResponseError)
+    )
+  }
+
+  def UnsupportedFormat: AddressBySearchResponseContainer = {
+    AddressBySearchResponseContainer(
+      response = Error,
+      status = BadRequestAddressResponseStatus,
+      errors = Seq(FormatNotSupportedAddressResponseError)
+    )
+  }
+
+  def EmptySearch: AddressBySearchResponseContainer = {
+    AddressBySearchResponseContainer(
+      response = Error,
+      status = BadRequestAddressResponseStatus,
+      errors = Seq(EmptyQueryAddressResponseError)
+    )
+  }
+
+  def Error: AddressBySearchResponse = {
```

Comment:
You don't override the value in the class that extends this trait, so everything that is not overwritten can be `val` in a trait.